File: src/Functional/Scala/patterns/PeerToPeer.scala
import scala.collection.mutable
object PeerToPeerExample { def run:Boolean={val b=mutable.ArrayBuffer[String]();val c=mutable.ArrayBuffer[String]();def send(f:String,t:String,d:String,i:mutable.ArrayBuffer[String])=i+=s"$f>$t:$d";send("peer-a","peer-b","block-42",b);send("peer-a","peer-c","block-42",c);(b++c).mkString(">")=="peer-a>peer-b:block-42>peer-a>peer-c:block-42"} }

File: src/Enterprise/C#/patterns/PeerToPeer.cs
using System.Collections.Generic;
namespace Genkidama.PatternExamples;
public static class PeerToPeerExample { public static bool Run(){var inbox=new List<string>();void Send(string f,string t,string d)=>inbox.Add($"{f}>{t}:{d}");Send("peer-a","peer-b","block-42");Send("peer-a","peer-c","block-42");return string.Join('>',inbox)=="peer-a>peer-b:block-42>peer-a>peer-c:block-42";} }
